core/src/main/scala/org/apache/spark/sql/hive/SnappyHiveExternalCatalog.scala

```diff
@@ -365,6 +375,23 @@ class SnappyHiveExternalCatalog private[hive](val conf: SparkConf,
     registerCatalogSchemaChange(refreshRelations)
   }
 
+  def dropTableUnsafe(schema: String, table: String, ignoreException: Int): Unit = {
+    try {
+      val catalogTable = super.getTable(schema, table)
+      // no exception raised while getting catalogTable
+      if (ignoreException == 1) {
+        withHiveExceptionHandling(super.dropTable(schema, table, true, true))
+      } else {
+        throw StandardException.newException(
+          SQLState.LANG_UNEXPECTED_USER_EXCEPTION, null, "Table retrieved successfully. To " +
+              "continue to drop this table change ignoreException argument in procedure to true");
+      }
+    } catch {
+      case a: AnalysisException =>
+        withHiveExceptionHandling(super.dropTable(schema, table, true, true))
+    }
```

Comment:
Add a case for other exceptions with appropriate message. Also match the message as well for AnalysisException